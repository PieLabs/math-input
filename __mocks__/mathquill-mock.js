const sinon = require('sinon');

const stubField = () => ({
  latex: sinon.stub()
});


const staticField = stubField();
const mathField = stubField();

const MQ = {
  StaticMath: () => staticField,
  MathField: () => mathField
}

const defaultExport = {
  getInterface: () => MQ,
  staticField,
  mathField
};

module.exports = defaultExport
