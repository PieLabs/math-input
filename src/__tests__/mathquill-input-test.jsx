import MathQuillInput from '../mathquill-input.jsx';
import React from 'react';
import { assert } from 'sinon';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { staticField } from '../../__mocks__/mathquill-mock';

const handleClick = () => { };

test('mounting', () => {
  const input = shallow(<MathQuillInput
    readOnly={false}
    latex={'1+1=2'}
    onClick={handleClick} />
  );

  // console.log('staticField: ', staticField);

  assert.calledWith(staticField.latex, '1+1=2');
});

test('MathQuillInput readonly false', () => {

  const notReadOnly = renderer.create(
    <MathQuillInput
      readOnly={false}
      latex={'1+1=2'}
      onClick={handleClick}
    />
  );

  let tree = notReadOnly.toJSON();
  expect(tree).toMatchSnapshot();
});

test('MathQuillInput readonly true', () => {

  const notReadOnly = renderer.create(
    <MathQuillInput
      readOnly={true}
      latex={'1+1=2'}
      onClick={handleClick}
    />
  );

  let tree = notReadOnly.toJSON();
  expect(tree).toMatchSnapshot();
});