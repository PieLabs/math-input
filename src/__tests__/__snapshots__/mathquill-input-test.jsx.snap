// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MathQuillInput readonly false 1`] = `
<div
  onClick={[Function]}
>
  <div
    className=""
  >
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    />
  </div>
  <div
    className="MathQuillInput-hidden-1"
  >
    <span />
  </div>
</div>
`;

exports[`MathQuillInput readonly true 1`] = `
<div
  onClick={[Function]}
>
  <div
    className="MathQuillInput-hidden-1"
  >
    <span
      onBlur={[Function]}
      onFocus={[Function]}
    />
  </div>
  <div
    className=""
  >
    <span />
  </div>
</div>
`;
